dev-docs/design/how-lit-html-works.md

```diff
@@ -0,0 +1,429 @@
+# Life of a lit-html render
+
+## About this Document
+
+This is an overview for how lit-html works, what makes it fast, and how the code is organized. It's intended for contributors, or anyone who is interested in the inner workings of lit-html. It is not a tutorial or introduction for Lit. For introductory material see the [Intro to Lit Tutorial](https://lit.dev/tutorials/intro-to-lit/), or [Using lit-html standalone](https://lit.dev/docs/libraries/standalone-templates/) documentation.
+
+All the source code described by this document can be found in [`lit-html.ts`](https://github.com/lit/lit/blob/main/packages/lit-html/src/lit-html.ts).
+
+If anything is unclear or you have questions, reach out on our [Discord channel](https://discord.com/invite/buildWithLit), or ask a question on [GitHub Discussions](https://github.com/lit/lit/discussions). All our communities can be found on the [Lit Community page](https://lit.dev/docs/resources/community/).
+
+# What is lit-html?
+
+lit-html is an HTML templating library. Templates are written in JavaScript by mixing static HTML strings and dynamic JavaScript values using template literals. lit-html enables UI as a function of application state programming model, fast initial rendering, and fast updates that minimally update DOM when state changes.
```

Comment:
> lit-html enables UI as a function of application state programming model

I think this isn't quite grammatically right, but the simple fixes are awkward (`enables **a** UI as function of`, or `enables **the** UI as function of state).

What about: `lit-html enables writing UI as a function of application state, it has fast initial rendering, and fast updates that:`

Reply:
Nice! Applied!